src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -77,12 +89,28 @@ public async Task<bool> IsEnabledAsync(string feature)
                             Parameters = featureFilterSettings.Parameters 
                         };
 
-                        if (await filter.EvaluateAsync(context).ConfigureAwait(false))
+                        //
+                        // IFeatureFilter
+                        if (filter is IFeatureFilter featureFilter && await featureFilter.EvaluateAsync(context).ConfigureAwait(false))
```

Comment:
>IFeatureFilter [](start = 38, length = 14)

we should check useAppContext first in case a filter instance implement both IFeatureFilter and IContextualFeatureFilter

Reply:
This is actually intentional, even if appContext is usable we still want to run IFeatureFilter if they implement that interface.

---
In reply to: [328391708](https://github.com/microsoft/FeatureManagement-Dotnet/pull/14#discussion_r328391708) [](ancestors = 328391708)